pkg/query/querier.go

```diff
@@ -18,6 +18,69 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
+var optCtxKey = struct{}{}
+
+// PartialErrReporter allows to report partial errors. Partial error occurs when only part of the results are ready and
+// another is not available because of the failure. We still want to return partial result, but with some notification.
+type PartialErrReporter interface {
+	// Report reports partial error (that the result will be only partial because given errors).
+	// NOTE: It is required to be thread-safe.
+	Report(err error)
+}
```

Comment:
Do we need an extra type or can the context just contain `report(error)` func?

Reply:
mhm, yeah single function, sounds good for this use case